stan/math/prim/scal/prob/normal_rng.hpp

```diff
@@ -1,39 +1,66 @@
 #ifndef STAN_MATH_PRIM_SCAL_PROB_NORMAL_RNG_HPP
 #define STAN_MATH_PRIM_SCAL_PROB_NORMAL_RNG_HPP
 
-#include <boost/random/normal_distribution.hpp>
-#include <boost/random/variate_generator.hpp>
 #include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
 #include <stan/math/prim/scal/err/check_finite.hpp>
-#include <stan/math/prim/scal/err/check_not_nan.hpp>
-#include <stan/math/prim/scal/err/check_positive.hpp>
-#include <stan/math/prim/scal/fun/constants.hpp>
-#include <stan/math/prim/scal/fun/value_of.hpp>
+#include <stan/math/prim/scal/err/check_positive_finite.hpp>
+#include <stan/math/prim/scal/meta/VectorBuilder.hpp>
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
+#include <stan/math/prim/scal/meta/max_size.hpp>
+#include <boost/random/normal_distribution.hpp>
+#include <boost/random/variate_generator.hpp>
 #include <string>
 
 namespace stan {
   namespace math {
-
-    template <class RNG>
-    inline double
-    normal_rng(double mu,
-               double sigma,
+    /**
+     * Return a pseudorandom Normal variate for the given location and scale
```

Comment:
Everything is pseudorandom here.  I only used `_rng` on the suffixes to keep things short and not confuse our users.